File: python/2010/grown-ups-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Grown Ups to the database
movies.insert(
    title="Grown Ups", 
    year=2010, 
    plot="After their high school basketball coach passes away, five good friends and former teammates reunite for a Fourth of July holiday weekend.", 
    rating=6
)

# Confirm that the movie was added
movie = movies.select(
    title="Grown Ups", 
    year=2010
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    

File: csharp/2010/grown-ups-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Grown Ups" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Grown Ups",
                year: 2010,
                plot: "After their high school basketball coach passes away, five good friends and former teammates reunite for a Fourth of July holiday weekend.",
                rating: 6
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Grown Ups",
                year: 2010
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}
